### DISCUSSION TITLE
REPL, Cider

### DISCUSSION THREAD

**Discussion by stevemolitor** [squint-cljs/cherry#107] [General]
Hi Clojure newbie here. Cherry is a really fun way for Javascripters like me to start learning Clojure, so thanks for this project.

Is it possible to get a Cherry REPL going with Cider in Emacs? If so how would one go about that? 

I'm just doing node.js scripting with Cherry atm; no browser stuff. 

Thanks!

Steve

**Comment by borkdude** [comment]
Hey @stevemolitor ! Yeah, we need a REPL / nREPL integration.

Note that nbb (https://github.com/babashka/nbb) has this but cherry doesn't have it yet.
One difficulty is that cherry emits ES6 modules/code by default and ES6 modules are pretty difficult to work with when it comes to REPL/re-defining vars, since modules are pretty much immutable after they have been defined.

I'm considering emitting just global vars in development mode, e.g.:

``` clojure
(ns foo)
(defn bar [x])
```

=>

```
globalThis.foo = {}
foo.bar = function(x);
```

instead of

```
function foo (x) {}
export x;
```

**Reply by stevemolitor** [reply]
Thanks @borkdude makes sense.

Setting the REPL aside, which would you recommend for writing Node.js scripts, nbb or cherry? Do you foresee cherry eventually replacing nbb, or is there a role for both? (Sorry if this is a FAQ.)

**Reply by borkdude** [reply]
I hope cherry can eventually replace nbb

**Reply by stevemolitor** [reply]
Related, I asked about squint + cider [here](https://github.com/squint-cljs/squint/discussions/195#discussioncomment-6750887).
